CollAction/ValidationAttributes/DateWithinMonthsAttribute.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+
+namespace CollAction.ValidationAttributes
+{
+    public class DateWithinMonthsAttribute : ValidationAttribute
+    {
+        private readonly int _months;
+
+        public String OfDateProperty { get; set; }
```

Comment:
I'd use two validation attributes for this. For comparing to another date, against comparing to the current date. They're different things. Separation of concerns.